packages/devtools_app/lib/src/service_manager.dart

```diff
@@ -441,6 +446,24 @@ class IsolateManager extends Disposer {
   final _mainIsolate = ValueNotifier<IsolateRef>(null);
   ValueListenable<IsolateRef> get mainIsolate => _mainIsolate;
 
+  Future<void> init(List<IsolateRef> isolates) async {
+    // Re-initialize isolates when VM developer mode is enabled/disabled to
+    // display/hide system isolates.
+    addAutoDisposeListener(preferences.vmDeveloperModeEnabled, () async {
+      final vmDeveloperModeEnabled = preferences.vmDeveloperModeEnabled.value;
+      final vm = await serviceManager.service.getVM();
+      final isolates = [
+        ...vm.isolates,
+        if (vmDeveloperModeEnabled) ...vm.systemIsolates,
+      ];
+      if (selectedIsolate.isSystemIsolate && !vmDeveloperModeEnabled) {
+        selectIsolate(_isolates.first.id);
+      }
```

Comment:
does this also need to be called outside of the listener callback before _initIsolates(isolates) on l:464?

Reply:
Nope, this should only be necessary when we remove system isolates from the list of available isolates. This just makes sure that we change the selected isolate to one of the remaining isolates if we currently have a system isolate selected when VM developer mode is disabled.